rskj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -461,32 +478,74 @@ public void createContract(DataWord value, DataWord memStart, DataWord memSize)
         // Start tracking repository changes for the constructor of the contract
         Repository track = getStorage().startTracking();
 
+        ProgramResult programResult = ProgramResult.empty();
+
         //In case of hashing collisions, check for any balance before createAccount()
-        if (track.isExist(newAddress)) {
-            Coin oldBalance = track.getBalance(newAddress);
-            track.createAccount(newAddress);
-            track.addBalance(newAddress, oldBalance);
+        if (track.isExist(contractAddress)) {
+            // Hashing collisions in CREATE are rare, but in CREATE2 are possible
+            // We check for the nonce to be zero and that the code to be empty
+            // if any of this conditions fails, the contract creation fails:
+            // We set an exception as the program result and set the program
+            // to empty so it does not executes
+
+
+            byte[] code = track.getCode(contractAddress);
+
+            if (getActivations().isActive(ConsensusRule.RSKIP125) && ((code == null || code.length == 0) ||
+                track.getNonce(contractAddress).compareTo(BigInteger.ZERO) != 0)) {
+                programResult.setException(ExceptionHelper.addressCollisionException(contractAddress));
+                programCode = new byte[]{};
+            }
```

Comment:
execution continues here why not aborting immediately?

Reply:
Good question, the EIP contradicts itself a little in that part. On one hand it says it should throw immediately, on the other that it should fail...
> with exactly the same behavior as would arise if the first byte in the init code were an invalid opcode 

So what I did was force the same path that it would follow for this error.

Looking in different Ethereum implementations I saw that they approach this in different ways, ethereumJ does what we do (force the same error and continuing the execution), geth fails immediately and so does aleth. 

I think you are right and failing immediately is the right thing to do, will do some tests and upload the fixed implementation.